### PR TITLE
Split out a separate endpoint to complete SSO registration

### PR DESCRIPTION
**Pull request by richvdh**
There are going to be a couple of paths to get to the final step of SSO reg, and I want the URL in the browser to consistent. So, let's move the final step onto a separate path, which we redirect to.

There are a couple of preparatory commits here, which should be reviewed separately.

### CODE REVIEW

File: changelog.d/9262.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve the user experience of setting up an account via single-sign on.
```
Comment (richvdh) [inline changelog.d/9262.feature:1]: obviously, there is more to follow.

File: synapse/rest/synapse/client/sso_register.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def __init__(self, hs: "HomeServer"):
+        super().__init__()
+        self._sso_handler = hs.get_sso_handler()
+
+    async def _async_render_GET(self, request: Request) -> None:
+        try:
+            session_id = get_username_mapping_session_cookie_from_request(request)
+        except SynapseError as e:
+            logger.warning("Error fetching session cookie: %s", e)
+            self._sso_handler.render_error(request, "bad_session", e.msg, code=e.code)
```
Comment (anoadragon453) [inline synapse/rest/synapse/client/sso_register.py:48]: I don't think this function takes an arg `code`.

**Review by anoadragon453 — APPROVED** [review]
lgtm besides one quirk. Also yay for separate resources.

File: synapse/rest/synapse/client/sso_register.py (hunk 2 above)
Comment (anoadragon453) [reply synapse/rest/synapse/client/sso_register.py:48]: Nvm, was looking at `synapse-dinsic` :innocent:

**Pull request merged**